src/controllers/protocols.controller.ts

```diff
@@ -0,0 +1,35 @@
+import catchAsync from "../utils/catchAsync";
+import ApiError from "../utils/ApiError";
+import httpStatus from "http-status";
+import {
+  addProtocol,
+  addSurveysToProtocol,
+  fetchSurveyIdByName,
+} from "../models/protocols.models";
+
+const createProtocol = catchAsync(async (req, res) => {
+  const { name, protocolData } = req.body;
+  const protocolId = (await addProtocol(name))[0].id;
+  if (protocolId) {

```

Comment:
also instead of adding and if else, and an if after the `await addProtocol` that checks if protocolId  is not truthy then send error, the code will look a little cleaner